src/server/game/Entities/Player/Player.cpp

```diff
@@ -22138,6 +22138,34 @@ void Player::SendCooldownEvent(SpellInfo const* spellInfo, uint32 itemId /*= 0*/
     data << uint32(spellInfo->Id);
     data << uint64(GetGUID());
     SendDirectMessage(&data);
+    uint32 cat = spellInfo->GetCategory();
+    uint32 catRec = spellInfo->CategoryRecoveryTime;
+    if (cat && catRec)
+    {
+        SpellCategoryStore::const_iterator ct = sSpellsByCategoryStore.find(cat);
+        if (ct != sSpellsByCategoryStore.end())
+        {
+            SpellCategorySet const& catSet = ct->second;
+            for (SpellCooldowns::const_iterator i = m_spellCooldowns.begin(); i != m_spellCooldowns.end(); ++i)
+            {
+                if (i->first == spellInfo->Id) // skip main spell, already handled above
+                    continue;
+
+                SpellInfo const* spellInfo2 = sSpellMgr->GetSpellInfo(i->first);
```

Comment:
Just leaving it as a note as @w1sht0l1v3  commented on the issue way long ago, this returns null pointers sporadically so it needs to be checked against that. Otherwise you may experience random crashes.